pytorch_lightning/overrides/distributed.py

```diff
@@ -104,6 +106,103 @@ def __iter__(self) -> Iterator[List[int]]:
         return iter(indices)
 
 
+class _DatasetFromSampler(Dataset):
+    """Dataset to create indexes from `Sampler`.
+
+    Adapted from https://github.com/catalyst-team/catalyst/blob/v22.04/catalyst/data/dataset.py#L6
```

Comment:
have we clarified the ownership, how significant is the adaptation?
cc: @carmocca